### PR TITLE
Disables feature from mono users that is unsupported by it and causes crash

### PR DESCRIPTION
**Pull request by oappi**
Solves issue  https://github.com/Jessecar96/SteamBot/issues/753    disables unsupported feature

### CODE REVIEW

**Comment by oappi** [comment]
Someone with more c# experience can try to actually fix it. This will just handle it gracefully

**Comment by scholtzm** [comment]
Indentation is off. You should use 4 spaces for tabs.

File: SteamBot/BotManager.cs [hunk 1]
```diff
@@ -49,10 +49,18 @@ public bool LoadConfiguration(string configFile)
 
             if (ConfigObject == null)
                 return false;
-
+            
+            mainLog = new Log(ConfigObject.MainLog, null, Log.LogLevel.Debug, Log.LogLevel.Debug);
+           
+            if (IsRunningOnMono () && ConfigObject.UseSeparateProcesses)  //mono to version 3.8 does not seem to support this
```
Comment (scholtzm) [inline SteamBot/BotManager.cs]: The extra space in front of `()` should be removed.

**Comment by oappi** [comment]
Sorry about that. Very new to c# and thought IDE automatically picked right indentation (used other languages though).

**Comment by oappi** [comment]
Should be 4 spaces now. Also fixed that space "IsRunningOnMono ()".  Wasn't sure if that was needed in c# so i followed Mono's documentations approach as examples had that space. http://www.mono-project.com/docs/gui/winforms/porting-winforms-applications/   (around half way of the page)

File: SteamBot/BotManager.cs [hunk 2]
```diff
@@ -49,10 +49,18 @@ public bool LoadConfiguration(string configFile)
 
             if (ConfigObject == null)
                 return false;
-
+            
+            mainLog = new Log(ConfigObject.MainLog, null, Log.LogLevel.Debug, Log.LogLevel.Debug);
+           
+            if (IsRunningOnMono() && ConfigObject.UseSeparateProcesses)  //mono to version 3.8 does not seem to support this
+			{
```
Comment (scholtzm) [inline SteamBot/BotManager.cs]: Indentation is still incorrect here.

File: SteamBot/BotManager.cs [hunk 3]
```diff
@@ -49,10 +49,18 @@ public bool LoadConfiguration(string configFile)
...
+			    mainLog.Info("Mono detected with UseSeparateProcesses as true, disabling" );
```
Comment (BlueRaja) [inline SteamBot/BotManager.cs]: Please make this a warning so it shows up in the console

**Comment by BlueRaja** [comment]
If you're using Visual Studio 2013 Community Edition *(which is free)*, `tools --> options --> text editor --> C# --> tabs`, set it to `indenting = smart` and `tab = insert spaces`.  I believe those are the default options..

**Comment by scholtzm** [comment]
The indentation is still incorrect. Uhh... :grinning:

**Comment by BlueRaja** [comment]
Also this is a good time to learn how [rebase](http://stackoverflow.com/questions/2715085/rebasing-and-what-does-one-mean-by-rebasing-pushed-commits) works :)

**Comment by BlueRaja** [comment]
Also, you should enable `Edit --> Advanced --> View Whitespace` to see your whitespace characters.

Even if you don't use Visual Studio for some reason, every half-decent text-editor I know of has a similar feature.

**Comment by oappi** [comment]
eh =/  cannot use visual studio atm. It would require windows uninstall, and i would rather not do that until (stable) w10 is out. Long story short i cannot install visual studio bc i installed it to partition that no longer exists, and no matter what i tried i couldn't get visual studio wiped out or repaired and that piece of cr.... wont allow me to reinstall it or get newer one  before i remove old  one that i don't really have)
Any change someone could fix those indentation... i dont really need the "fame" which already starts to be rather questionable. github's editor screws those indentation  even more. Thanks for the rebase link

**Comment by oappi** [comment]
any change someone could sort those indentation the way you want them? For some reason every time i check it in my own fork there are those white spaces instead of tabs  even though they are tabs in my notepad++  :rage:   and in github editor even the "original" has white spaces instead of tabs (at least for me), and github editor wont save change from 8 space tab to 4 space tab.

fear not. Wont be sending more code as long as this project is in C#

**Comment by BlueRaja** [comment]
...I think you are confusing what whitespace and tabs are.  This project _(and most other projects these days, in most other languages)_ prefer tabs to be converted to spaces.  You do **not** want tabs in your code.

**Comment by oappi** [comment]
i guess we have different priorities. i  introduced what crashed your bot even though i didn't even get confirmation it was a bug, but was still able to find problem and at least handle it gracefully. I was happy enough to finally get my altered bot from windows client to linux   and wanted to share solution even though i don't currently have decent editor bc of how visual studios installation works. I dont need the "credit" just wanted to give something back for what i got out of the freely available code. As i said, fear not. Wont be troubling you anymore with my whitespaces.

**Comment by oappi** [comment]
@WildCard65 Would be more efficient, but then again, unless i misunderstood something (which is highly likely) that block  is only run once and the idea for if blocks was to inform user that feature requested in setting file has been disabled due to limitation of mono. Additionally it also makes  finding that if clause easier  when mono's feature set support .net's (4.5.1?) feature set required by steambot. Surely for personal use your solution is better since you know it is there.

**Comment by Jessecar96** [comment]
tabs > spaces

**Comment by BlueRaja** [comment]
@oappi: Do you plan on cleaning up this PR, or should it be closed?

**Comment by Jessecar96** [comment]
This looks abandoned, closing now. Make a new PR if you want to add this in the future.

**Pull request closed without merging**